src/crafting.cpp

```diff
@@ -993,82 +993,137 @@ bool Character::craft_proficiency_gain( const item &craft, const time_duration &
     return player_gained_prof;
 }
 
-double Character::crafting_success_roll( const recipe &making ) const
+float Character::get_recipe_weighted_skill_average( const recipe &making ) const
 {
     if( has_trait( trait_DEBUG_CNF ) ) {
         return 1.0;
     }
-    int secondary_dice = 0;
+    int secondary_skill_total = 0;
     int secondary_difficulty = 0;
-    for( const auto &pr : making.required_skills ) {
-        secondary_dice += get_skill_level( pr.first );
-        secondary_difficulty += pr.second;
+    for( const auto &count_secondaries : making.required_skills ) {
+        // the difficulty of each secondary skill, count_secondaries.second, adds weight:
+        // skills required at a higher level count more.
+        secondary_skill_total += get_skill_level( count_secondaries.first ) * count_secondaries.second;
+        secondary_difficulty += count_secondaries.second;
     }
+    // The primary required skill counts extra compared to the secondary skills, before factoring in the
+    // weight added by the required level.
+    const float weighted_skill_average =
+        ( ( 2.0f * making.difficulty * get_skill_level( making.skill_used ) ) + secondary_skill_total ) /
+        ( 2.0f * making.difficulty + secondary_difficulty );
```

Comment:
If we want to make the fail multipliers on proficiencies into a real multiplier, here is where we'd add them, by dividing your `weighted_skill_average` by `recip.fail_multiplier`.